tests/component/_task_modules/test_in_stream_read_properties.py

```diff
@@ -126,7 +128,7 @@ def test___ai_task___reset_bool_property___returns_default_value(ai_task: Task):
     raises=DaqError,
 )
 def test___ai_task___get_string_property___returns_default_value(ai_task: Task):
-    assert ai_task.in_stream.logging_file_path == ""
+    assert ai_task.in_stream.logging_file_path == pathlib.Path()
```

Comment:
Sigh, I'm not sure we want this to return a path that is equivalent to "." (current directory).

Reply:
@mshafer-NI do you have any thoughts on returning "." path as default value? It seems like [pathlib](https://docs.python.org/3/library/pathlib.html#methods-and-properties) does not have property to be set as "not a path".

Reply:
I normally use `None` for "not a path" -> in this case, it probably makes sense for the default to be `pathlib.Path.cwd()`

Reply:
> I normally use None for "not a path"

Ok, so the property setter should accept `Optional[Union[str, pathlib.PurePath]]` and the property getter should return `Optional[pathlib.Path]`?

> -> in this case, it probably makes sense for the default to be pathlib.Path.cwd()

Why? This property is a file path, not a directory path.

Reply:
Would it be better to keep returning `str` from the property getter, but accept `Union[str, pathlib.PurePath]` in the property setter?

Reply:
> Why? This property is a file path, not a directory path.

Oh, I missed that, yeah, probably return None

> Would it be better to keep returning str from the property getter, but accept Union[str, pathlib.PurePath] in the property setter?

Our guidance is to use pathlib (because it's nicer) unless you are in a _very_ heavily executed code path where swithing to using string operations can save a measurable amount of time.

Reply:
Thanks for the inputs, I will change it to return None if the value is empty string; else return file path.